chopsticks/tunnel.py

```diff
@@ -679,21 +679,25 @@ class SSHTunnel(SubprocessTunnel):
     :param host: The hostname to connect to, as would be specified on an
                  ``ssh`` command line.
     :param user: The username to connect as.
+    :param port: A custom port other than 22 to connect to.
     :param sudo: If true, use ``sudo`` on the remote end in order to run as
                  the ``root`` user. Use this when you can ``sudo`` to root but
                  not ``ssh`` directly as the root user.
 
     """
-    def __init__(self, host, user=None, sudo=False):
+    def __init__(self, host, user=None, port=None, sudo=False):
         self.host = host
         self.user = user
+        self.port = str(port)
```

Comment:
Let's not stringify the port here. It's cleaner to keep it as an integer until we want to construct the command line.

While I wouldn't document this as recommended behaviour, this would allow re-assigning a port on a tunnel naturally:

```
tunnel.close()
tunnel.port = 2222
tunnel.connect()
```